module/applications/actor/trait-selector.mjs

```diff
@@ -133,18 +148,24 @@ export default class TraitSelector extends BaseConfigSheet {
 
   /**
    * Filter a list of choices that begin with the provided key for update.
-   * @param {string} prefix    They initial form prefix under which the choices are grouped.
+   * @param {string} prefix    The initial form prefix under which the choices are grouped.
    * @param {string} path      Path in actor data where the final choices will be saved.
    * @param {object} formData  Form data being prepared. *Will be mutated.*
    * @protected
    */
   _prepareChoices(prefix, path, formData) {
-    const chosen = [];
+    const chosen = new Set();
     for ( const key of Object.keys(formData).filter(k => k.startsWith(`${prefix}.`)) ) {
-      if ( formData[key] ) chosen.push(key.replace(`${prefix}.`, ""));
+      if ( formData[key] ) chosen.add(key.replace(`${prefix}.`, ""));
       delete formData[key];
     }
-    formData[path] = chosen;
+
+    // Add choices from the source that have been removed by an override: if we didn't, the override would be persisted
+    const source = new Set(foundry.utils.getProperty(this.document._source, path));
+    const current = foundry.utils.getProperty(this.document, path);
+    for ( const choice of source.difference(current) ) chosen.add(choice);
+
+    formData[path] = Array.from(chosen).sort((a, b) => a.localeCompare(b, "en"));
```

Comment:
Should this be `game.i18n.lang`?

```suggestion
    formData[path] = Array.from(chosen).sort((a, b) => a.localeCompare(b, game.i18n.lang));
```

Reply:
Since this is sorting the `keys`, I'm not sure it would make a difference. Could probably even just use a simple `Array.from(chosen).sort()` here since the specific order is unimportant, only the consistency of the sorting.